File: javascript/programmers/javascript/level1/콜라츠추측.js
const n = 626331;

function solution(num) {
  let answer = 0;
  while (num != 1 && answer != 500) {
    if (num % 2 === 0) num /= 2;
    else num = num * 3 + 1;
    answer++;
  }
  if (answer === 500) answer = -1;
  return answer;
}

console.log(solution(n));

File: python/programmers/level1/콜라츠추측.py
def solution(num):
    count = 0
    while count <= 500 and num != 1:
        if num % 2 == 0:
            num //= 2
        else:
            num = num * 3 + 1
        count += 1
    if count == 501:
        return -1
    return count
